File: core/src/jvmMain/java/de/haukesomm/sokoban/core/level/LevelCharacterMap.java
package de.haukesomm.sokoban.core.level;

import de.haukesomm.sokoban.core.*;

import java.util.Optional;

public interface LevelCharacterMap {

    TileType getTileType(char character);

    Optional<EntityType> getEntityType(char character);
}

File: core/src/commonMain/kotlin/de/haukesomm/sokoban/core/level/LevelCharacterMap.kt
package de.haukesomm.sokoban.core.level

import de.haukesomm.sokoban.core.EntityType
import de.haukesomm.sokoban.core.TileType

interface LevelCharacterMap {

    fun getTileType(character: Char): TileType?

    fun getEntityType(character: Char): EntityType?
}
